File: festival-web/src/main/java/cn/bdqfork/web/route/message/DefaultHttpMessageHandler.java
package cn.bdqfork.web.route.message;

import cn.bdqfork.web.route.message.resolver.ParameterResolver;
import cn.bdqfork.web.route.message.resolver.ParameterResolverFactory;
import io.vertx.reactivex.ext.web.RoutingContext;

import java.lang.reflect.Parameter;
import java.util.ArrayList;
import java.util.List;

/**
 * @author bdq
 * @since 2020/1/31
 */
public class DefaultHttpMessageHandler extends AbstractHttpMessageHandler {
    private ParameterResolver parameterResolver;

    public DefaultHttpMessageHandler(ParameterResolverFactory parameterResolverFactory) {
        this.parameterResolver = parameterResolverFactory.createResolverChain();
    }

    @Override
    protected Object[] doHandle(RoutingContext routingContext, Parameter[] parameters) {

        List<Object> args = new ArrayList<>(parameters.length);

        for (Parameter parameter : parameters) {
            Object value = parameterResolver.resolve(parameter, routingContext);
            args.add(value);
        }
        return args.toArray();
    }

}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/message/DefaultHttpMessageHandler.kt
package cn.bdqfork.web.route.message

import cn.bdqfork.web.route.message.resolver.ParameterResolver
import cn.bdqfork.web.route.message.resolver.ParameterResolverFactory
import io.vertx.ext.web.RoutingContext
import java.lang.reflect.Parameter
import java.util.*

/**
 * @author bdq
 * @since 2020/1/31
 */
class DefaultHttpMessageHandler(parameterResolverFactory: ParameterResolverFactory) : AbstractHttpMessageHandler() {
    private val parameterResolver: ParameterResolver = parameterResolverFactory.createResolverChain()
    override fun doHandle(routingContext: RoutingContext, parameters: Array<Parameter>): Array<Any?> {
        val args: MutableList<Any?> = ArrayList(parameters.size)
        for (parameter in parameters) {
            val value = parameterResolver.resolve(parameter, routingContext)
            args.add(value)
        }
        return args.toTypedArray()
    }

}